universe/Effect.h

```diff
@@ -765,6 +765,50 @@ class FO_COMMON_API AddSpecial : public EffectBase {
     void serialize(Archive& ar, const unsigned int version);
 };
 
+/** Removes all buildings from the target object. */
+class FO_COMMON_API RemoveAllBuildings : public EffectBase {
```

Comment:
This seems unnecessary. There is already a destroy effect that can be targeted to all buildings on a planet.

Reply:
If I'm not mistaken, this requires a new scope for the buildings on the planet. With this solution, the script is shorter and easier to read.

Reply:
Use `Effect::Conditional` to narrow the target set when executing an effectsgroup.

Reply:
Can you give an example how to do that?